src/main/kotlin/com/sourcegraph/cody/edit/EditCodeInlayRenderer.kt

```diff
@@ -0,0 +1,61 @@
+package com.sourcegraph.cody.edit
+
+import com.intellij.openapi.editor.EditorCustomElementRenderer
+import com.intellij.openapi.editor.Inlay
+import com.intellij.openapi.editor.markup.TextAttributes
+import com.intellij.ui.JBColor
+import java.awt.Graphics
+import java.awt.Rectangle
+import javax.swing.CellRendererPane
+import javax.swing.JPanel
+
+class EditCodeInlayRenderer(private val component: JPanel, private val requiredHeight: Int) :
+    EditorCustomElementRenderer {
+
+  private val rendererPane = CellRendererPane()
+
+  init {
+    // Call validate() initially to ensure the layout is updated
+    rendererPane.add(component)
+    component.validate()
```

Comment:
General comment so I won't forget later:
be careful with EDT calls in init, as then every object creation of `EditCodeInlayRenderer` need to be in the EDT as well, and in past we had hard to spot bugs because of this.